x-pack/plugins/infra/public/pages/metrics/hosts/components/truncate_link_with_tooltip.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiToolTip, EuiLink } from '@elastic/eui';
+import React from 'react';
+import { euiStyled } from '@kbn/kibana-react-plugin/common';
+import { LinkDescriptor, useLinkProps } from '@kbn/observability-plugin/public';
+
+const tooltipAnchorClassname = '_infra_truncate_tooltip_anchor_';
+
+const TooltipWrapper = euiStyled.div`
+  width: 100%;
+  .${tooltipAnchorClassname} {
+    width: 100% !important;
+    display: block !important;
+  }
+`;
+
+const StyledLink = euiStyled(EuiLink)`
+  display: block;
+  max-width: 100%;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+interface Props {
+  text: string;
+  linkProps: LinkDescriptor;
+}
+
+export function TruncateLinkWithTooltip(props: Props) {
+  const { text, linkProps, ...rest } = props;
+
+  const link = useLinkProps(linkProps);
+
+  return (
+    <TooltipWrapper {...rest}>
+      <EuiToolTip delay="long" content={text} anchorClassName={tooltipAnchorClassname}>
+        <StyledLink {...link}>{text}</StyledLink>
```

Comment:
**question**: not sure if it's the recommended practice, but could you use the utility classes `eui-displayBlock` and `eui-textTruncate` from EUI to achieve the same result without creating the extended StyledLink component?
It should be something like
```jsx
<EuiLink className="eui-displayBlock eui-textTruncate" {...link}>
```